### PR TITLE
APP-4556 Expose Portal React-Select props

### PR DESCRIPTION
**Pull request by symphony-jean-michael**
**Jira ticket**
https://perzoinc.atlassian.net/browse/APP-4556

**Changes**
- Expose Portal props ('menuPortalStyles' , 'menuPortalTarget' and 'menuShouldBlockScroll')
- Move Dropdown types/interfaces in the interface file
- Add a "PortalTemplate" template and use it in the TimePicker and Dropdown stories

**Demo**

Screenshot 2021-10-12 at 10 34 44

Screenshot 2021-10-12 at 11 58 39

### CODE REVIEW

File: src/components/dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -17,149 +18,15 @@ import {
...
-  TagRendererProps,
 } from './interfaces';
-import { HasValidationProps } from '../validation/interfaces';
-import { HasTooltipProps } from '../tooltip/interfaces';
 import LabelTooltipDecorator from '../label-tooltip-decorator/LabelTooltipDecorator';
 
 // css baseclass prefix
 const prefix = 'tk-select';
 
-export type DropdownProps<T> = {
```
Comment (symphony-annaLlorens) [inline src/components/dropdown/Dropdown.tsx:32]: Nice refactor!

**Comment by symphony-annaLlorens** [comment]
On Nexus the were some specific methods to render the menu on top or bottom of the dropdown depending on the position of the Dropdown on the chat. I wonder if this will still work or it's compatible. 🙄

File: src/components/time-picker/TimePicker.tsx [hunk 2]
```diff
@@ -239,6 +242,9 @@ export const TimePicker: React.FC<TimePickerProps> = ({
       tooltip={tooltip}
       tooltipCloseLabel={tooltipCloseLabel}
       showRequired={showRequired}
+      menuPortalStyles={menuPortalStyles}
```
Comment (AliciaSymphony) [inline src/components/time-picker/TimePicker.tsx:245]: Could you keep the alphabetical order please :)

File: src/components/time-picker/TimePicker.tsx [hunk 3]
```diff
@@ -261,6 +267,9 @@ TimePicker.propTypes = {
   tooltipCloseLabel: PropTypes.string,
   showRequired: PropTypes.bool,
   value: PropTypes.string,
+  menuPortalStyles: PropTypes.object,
```
Comment (AliciaSymphony) [inline src/components/time-picker/TimePicker.tsx:270]: same comment

File: stories/Dropdown.stories.tsx [hunk 4]
```diff
@@ -245,6 +246,16 @@ export const Select: React.FC = () => (
...
 );
 
+export const Portal = PortalTemplate.bind({});
+Portal.args = {
+  title: 'A Dropdown Menu in a Portal',
+  component: <Dropdown
+    options={defaultOptions}
+    menuPortalTarget={document.body}
+    menuShouldBlockScroll={true}
+    menuPortalStyles={{ zIndex: 100 }} />,
```
Comment (AliciaSymphony) [inline stories/Dropdown.stories.tsx:256]: beside the zIndex, is there another use?

Comment (symphony-adnane) [reply stories/Dropdown.stories.tsx:256]: is it really necessary to adjust the zIndex or is it just for demo purpose?

**Review by symphony-adnane — APPROVED** [review]

**Pull request merged**